passes/techmap/clock_gate.cc

```diff
@@ -0,0 +1,205 @@
+#include "kernel/yosys.h"
+#include "kernel/ff.h"
+#include <optional>
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+struct ClockGateCell {
+	std::string name;
+	std::string ce_pin;
+	std::string clk_in_pin;
+	std::string clk_out_pin;
+
+	ClockGateCell(std::string& name, std::string& str) : name(RTLIL::escape_id(name)) {
+		char delimiter = ':';
+		size_t pos1 = str.find(delimiter);
+		if (pos1 == std::string::npos)
+			log_assert(false && "Not enough ports in descriptor string");
```

Comment:
It would be nice to have those be `log_cmd_error` as long as we are careful where from this constructor is called. I guess we should pull the parsing into a method of its own when it calls `log_cmd_error`.